data/completion/etelpmoc.sh

```diff
@@ -0,0 +1,195 @@
+#!/bin/bash
+
+# etelpmoc is the backwards half of complete: it de-serialises the tab
+# completion request into the appropriate environs expected by the tab
+# completion tools, performs whatever action is wanted, and serialises the
+# result. It accomplishes this by having functions override the builtin
+# completion commands.
+#
+# this always runs "inside", in the same environment you get when doing "snap
+# run --shell", and snap-exec is the one setting the first argument to the
```

Comment:
No please. I think it would be quite surprising for a completion script to by hitting the network or doing privileged operations outside of the snap. I strongly prefer the first iteration is 'no plugs allowed' for the completer and only after we have clear use cases do we consider something more.